### PR TITLE
Truncate long author lists

### LINKED ISSUE #22: Do something about lots of authors
Some items have lots and lots of authors. It's unlikely displaying them all is useful and it sure looks bad.

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This closes #22

### CODE REVIEW

**Comment by matt-bernhardt** [comment]
I'm still working through the most efficient way to call this. Right now it works for something like

```
foo = Result.new(...)
foo.authors = foo.truncate_authors(['foo','bar','baz','none'])
```

File: app/models/result.rb [hunk 1]
```diff
@@ -12,4 +12,11 @@ def initialize(title, year, url, type)
     @url = url
     @type = type
   end
+
+  # Testing something
+  def truncate_authors(authors)
+    return @authors if !ENV['MAX_AUTHORS'] || authors.length <= ENV['MAX_AUTHORS'].to_i
```
Comment (JPrevost) [inline app/models/result.rb]: Negative conditions are sometimes hard to grok. Rails allows `unless` to allow you to not have to `!` things.

Also, I think we can just make this a required ENV and not do that check at all.

**Comment by matt-bernhardt** [comment]
Ok, cool. I'll make those changes and adjust the readme.

**Comment by JPrevost** [comment]
When making use of new ENV, especially required ones like this, you'll want to update `app.json` to require it as well which tells the PR builds to copy that ENV setting from the Staging app.

File: test/models/normalize_eds_test.rb [hunk 2]
```diff
@@ -56,6 +56,19 @@ class NormalizeEdsTest < ActiveSupport::TestCase
     end
   end
 
+  test 'can handle many authors' do
```
Comment (JPrevost) [inline test/models/normalize_eds_test.rb]: This test is not really needed as the code you are adding operates on Results and thus doesn't care about how the authors were originally added to a Result in the first place.

File: test/models/normalize_worldcat_test.rb [hunk 3]
```diff
@@ -59,4 +59,16 @@ class NormalizeWorldcatTest < ActiveSupport::TestCase
       assert_equal(0, query['total'])
     end
   end
+
+  test 'searches with many authors get truncated list' do
```
Comment (JPrevost) [inline test/models/normalize_worldcat_test.rb]: Also not really necessary (see comment about test in `test/models/normalize_eds_test.rb`)

File: app/models/result.rb [hunk 4]
```diff
@@ -12,4 +12,10 @@ def initialize(title, year, url, type)
     @url = url
     @type = type
   end
+
+  # View-type method for returning a truncated list of authors.
+  def truncated_authors
+    return authors if authors.length <= ENV['MAX_AUTHORS'].to_i
+    authors[0..ENV['MAX_AUTHORS'].to_i - 1] << 'et al'
```
Comment (JPrevost) [inline app/models/result.rb]: A more ruby way to write this is:
`authors[0...ENV['MAX_AUTHORS'].to_i] << 'et al'`

See: http://ruby-doc.org/core-2.3.1/Range.html

**Pull request merged**